src/pkgcheck/checks/python.py

```diff
@@ -422,3 +426,196 @@ def feed(self, pkg):
                 if GITHUB_ARCHIVE_RE.match(uri):
                     yield PythonGHDistfileSuffix(f.filename, uri, pkg=pkg)
                     break
+
+
+class PythonHasVersionUsage(results.VersionResult, results.Style):
+    """Package uses has_version inside ``python_check_deps``.
+
+    An ebuild that declares ``python_check_deps`` function, which tests
+    python implementations for matching dependencies, should use the special
+    ``python_has_version`` for best log output and defaults [#]_.
+
+    .. [#] https://projects.gentoo.org/python/guide/any.html#dependencies
+    """
+
+    def __init__(self, lines, **kwargs):
+        super().__init__(**kwargs)
+        self.lines = tuple(lines)
+
+    @property
+    def desc(self):
+        s = pluralism(self.lines)
+        lines = ', '.join(map(str, self.lines))
+        return f'usage of has_version on line{s}: {lines}, replace with python_has_version'
+
+
+class PythonHasVersionMissingPythonUsedep(results.LineResult, results.Error):
+    """Package calls ``python_has_version`` or ``has_version`` without
+    ``[${PYTHON_USEDEP}]`` suffix.
+
+    All argument  to ``python_has_version`` or ``has_version`` inside
+    ``python_check_deps`` should contain ``[${PYTHON_USEDEP}]`` suffix for the
+    dependency [#]_.
+
+    .. [#] https://projects.gentoo.org/python/guide/any.html#dependencies
+    """
+
+    @property
+    def desc(self):
+        return f'line: {self.lineno}: missing [${{PYTHON_USEDEP}}] suffix for argument "{self.line}"'
+
+
+class PythonMismatchUseCheckVersionAnyDep(results.VersionResult, results.Warning):
```

Comment:
Something about the check name is hard to parse but I don't have a better suggestion yet. Did you have any alternatives?

Reply:
How about PythonAnyR1MismatchedUseHasVersionCheck?